evap/staff/importers/enrollment.py

```diff
@@ -0,0 +1,666 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, Iterable, Optional, Sequence, Set
+
+from django.conf import settings
+from django.db import transaction
+from django.utils.html import format_html
+from django.utils.translation import gettext as _
+
+from evap.evaluation.models import Contribution, Course, CourseType, Degree, Evaluation, Semester, UserProfile
+from evap.staff.tools import create_user_list_html_string_for_message
+
+from .base import (
+    Checker,
+    ConvertExceptionsToMessages,
+    ExcelFileLocation,
+    ExcelFileRowMapper,
+    FirstLocationAndCountTracker,
+    ImporterLog,
+    ImporterLogEntry,
+    InputRow,
+    RowCheckerMixin,
+)
+from .user import UserData, UserDataEmptyFieldsChecker, UserDataMismatchChecker, UserDataValidationChecker, UserImporter
+
+
+@dataclass
+class CourseData:
+    """Holds information about a course, retrieved from an import file."""
+
+    name_de: str
+    name_en: str
+    degrees: Set[Degree]
+    course_type: CourseType
+    is_graded: bool
+    responsible_email: str
+
+    def equals_when_ignoring_degrees(self, other):
+        return (
+            self.name_de == other.name_de
+            and self.name_en == other.name_en
+            and self.course_type == other.course_type
+            and self.is_graded == other.is_graded
+            and self.responsible_email == other.responsible_email
+        )
+
+
+@dataclass
+class PartialCourseData:
+    """Same as CourseData, but some fields might be unpopulated because parsing them failed"""
+
+    name_de: str
+    name_en: str
+    degrees_or_none: Set[Degree]
+    course_type_or_none: CourseType
+    is_graded_or_none: bool
```

Comment:
Hm, I suppose it would be clearer to make the types `Optional` instead of naming them `*_or_none`? (Or maybe do both); for me, I will see / find type hints relatively quickly so the `or_none` is more noise

edit: ah, I see, this is to distinguish it from the fully parsed data; hm I guess we can stick to this name

Reply:
Made them `Optional[X]`. I'm open for removing the `or_none` part of the name -- I think for developing, it might help people to distinguish where they are (and if this is optional or not), depending on what tools they have available.

Reply:
Yes, let's keep them